homeassistant/components/evohome/climate.py

```diff
@@ -149,7 +149,12 @@ def __init__(self, evo_broker, evo_device) -> None:
         """Initialize a Honeywell TCC Zone."""
         super().__init__(evo_broker, evo_device)
 
-        self._unique_id = evo_device.zoneId
+        if evo_device.modelType[:13] == "VisionProWifi":
+            # this system does not have a distinct ID for the zone
+            self._unique_id = f"{evo_device.zoneId}z"
```

Comment:
If it doesn't have a distinct ID, then is should not have a `unique_id` at all?

Reply:
Fair point - this is an older/esoteric piece of kit (Honeywell VisionPro Wifi), never seen one before - an edge case, but it was supported by this integration previously.

I did consider 'withdrawing' support for it, as I don't want to get committed to extensive writing workarounds like this for (maybe) 1-2 users...

This is what the JSON looks like for this kit:
```json
{
    "locationId": "2166142",
    "gateways": [{
            "gatewayId": "395968",
            "temperatureControlSystems": [{
                    "systemId": "416856",
                    "zones": [{
                            "zoneId": "416856",
                            "temperatureStatus": {

```
In all other systems I've seen, the `zoneId` value (416856) is distinct from the `systemId` - this one, they share the same value! This has come to a head due to several issues, one being that the controller and the zone are separate entities and the controller is implemented as a climate entity - hence the need for two unique IDs.

I am satisfied this is a special case, and adding the `z` suffix is a simple fix that will allow this system to behave as all the other evohome-compatible systems.

Reply:
BTW, the above ID numbers are not known to change.

Are you saying: 
```python
 if evo_device.modelType[:13] == "VisionProWifi":
            # this system does not have a distinct ID for the zone
            self._unique_id = None
```

What are the implications of that?

Reply:
No entity registry support. No device info.

Reply:
My vote is we leave it in - it is a unique ID for the zone (once that z is added),

Reply:
@zxdavb I've seen this pattern before with other thermostats that only have a single zone.  The manufacturer uses same ids for the thermostat/system and the zone when its a non-zoned system.

Maybe something like?
```
self._unique_id = f"{evo_device.zoneId}z" if evo_device.zoneId == evo_device.systemId else evo_device.zoneId
```